pkg/piecestore/psserver/server.go

```diff
@@ -178,15 +170,6 @@ func (s *Server) Piece(ctx context.Context, in *pb.PieceId) (*pb.PieceSummary, e
 		return nil, err
 	}
 
-	match, err := regexp.MatchString("^[A-Za-z0-9]{20,64}$", id)
```

Comment:
why was this part removed?

Reply:
this was removed because all pieceIDs are passed into the `getNamespacedPieceID` function which will return a base58 encoded string, so I thought there wasn't a need for this check, but I can go ahead and add it back.

Reply:
Ah, yeah... although I just realized there's a problem with this. It could have collisions. file-systems should be regarded as case-insensitive, whereas base58 is somewhat case sensitive. Yeah, let's keep it removed.